### PR TITLE
Support ChainerX in `Variable.addgrad`

### LINKED ISSUE #6983: Support ChainerX in Variable.addgrad
Required in MNIST data parallel example (#6857)

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6983

~Depends on #6982~

### CODE REVIEW

**Comment by niboshi** [comment]
Rebased. PTAL

File: chainer/variable.py [hunk 1]
```diff
@@ -1303,14 +1309,10 @@ def addgrad(self, var):
...
 
         dst = self.grad_var
-
-        src_dev = cuda.get_device_from_array(src.array)
-        dst_dev = cuda.get_device_from_array(self.array)
-
-        if src_dev.id != dst_dev.id:
-            src = chainer.functions.copy(src, dst_dev.id)
-        self._grad_var = src if dst is None else src + dst
-        self._grad = None
```
Comment (asi1024) [inline chainer/variable.py]: Could you explain why is this line removed?

**Comment by asi1024** [comment]
Jenkins, test this please.

File: chainer/variable.py (hunk 1 above)
Comment (niboshi) [reply chainer/variable.py]: `_grad` is set inside `grad_var.setter`.
(I don't know why previously `None` was always assigned).

Comment (asi1024) [reply chainer/variable.py]: I understand. Thanks.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins failure seems unrelated to this PR. LGTM.
